web/src/compose_paste.ts

```diff
@@ -60,6 +60,19 @@ export function is_white_space_pre(paste_html: string): boolean {
     );
 }
 
+function is_from_mac_terminal(html_fragment: HTMLBodyElement): boolean {
+    const html_tag = html_fragment.parentElement;
+    if (!html_tag || html_tag.nodeName !== "HTML") {
+        return false;
+    }
+
+    const has_mac_terminal_metadata = [...html_tag.querySelectorAll("meta")].some(
+        (meta) => meta.name === "Generator" && meta.content?.includes("Cocoa HTML Writer"),
+    );
```

Comment:
Can you add a test for this code path? It'll be helpful to see an example with whatever `meta.content` is, to consider possible improvements.